File: 1189_maximum-number-of-balloons/csharp/solution.cs
public class Solution
{
    public int MaxNumberOfBalloons(string text)
    {
        const int MAX_LETTERS = 26;
        const string TO_FIND = "balloon";

        int[] sCount = new int[MAX_LETTERS];
        int[] targetCount = new int[MAX_LETTERS];

        foreach (char c in text)
            sCount[c - 'a']++;
        foreach (char c in TO_FIND)
            targetCount[c - 'a']++;

        int result = int.MaxValue;

        for (int i = 0; i < MAX_LETTERS; i++)
        {
            if (targetCount[i] > 0)
            {
                result = Math.Min(result, sCount[i] / targetCount[i]);
            }
        }

        return result;
    }
}

File: 1189_maximum-number-of-balloons/rust/solution.rs
use std::cmp;

#[cfg(feature = "harness")]
pub struct Solution;

impl Solution {
    pub fn max_number_of_balloons(text: String) -> i32 {
        let max_letters = 26;
        let to_find = "balloon";
        let mut s_count = vec![0; max_letters];
        let mut target_count = vec![0; max_letters];

        for c in text.chars() {
            s_count[c as usize - 'a' as usize] += 1;
        }

        for c in to_find.chars() {
            target_count[c as usize - 'a' as usize] += 1;
        }

        let mut result = i32::MAX;

        for i in 0..max_letters {
            if target_count[i] > 0 {
                result = cmp::min(result, s_count[i] / target_count[i]);
            }
        }

        result
    }
}

